homeassistant/components/velbus/services.py

```diff
@@ -35,20 +36,34 @@ def check_entry_id(interface: str) -> str:
             "The interface provided is not defined as a port in a Velbus integration"
         )
 
+    def get_config_entry(interface: str) -> ConfigEntry | None:
+        for config_entry in hass.config_entries.async_entries(DOMAIN):
+            if "port" in config_entry.data and config_entry.data["port"] == interface:
+                return config_entry
+        return None
+
     async def scan(call: ServiceCall) -> None:
-        await hass.data[DOMAIN][call.data[CONF_INTERFACE]]["cntrl"].scan()
+        """Handle a scan service call."""
+        entry = get_config_entry(call.data[CONF_INTERFACE])
+        if entry:
+            await entry.runtime_data["cntrl"].scan()
```

Comment:
This looks wrong (and the ones below)

Reply:
how should i solve this in case the entry is not found?